presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/StreamPropertyDerivations.java

```diff
@@ -459,6 +459,9 @@ public StreamProperties visitTopNRowNumber(TopNRowNumberNode node, List<StreamPr
         @Override
         public StreamProperties visitTopN(TopNNode node, List<StreamProperties> inputProperties)
         {
+            if (node.getStep().equals(TopNNode.Step.PARTIAL)) {
+                return Iterables.getOnlyElement(inputProperties);
+            }
```

Comment:
I don't understand this commit (sure, it's more of a problem in myself), but could you consider come comment explaining this otherwise obvious change?